orderly_web/start.py

```diff
@@ -24,15 +25,24 @@ def start(path, extra=None, options=None, pull_images=False):
     cfg.resolve_secrets()
     if pull_images:
         pull(cfg)
+
+    notifier = Notifier(cfg.slack_webhook_url)
     with docker_client() as cl:
-        ensure_network(cl, cfg.network)
-        for v in cfg.volumes.values():
-            ensure_volume(cl, v)
-        orderly_init(cfg, cl)
-        web_init(cfg, cl)
-        proxy_init(cfg, cl)
-        config_save(cfg)
-        return True
+        notifier.post("*Starting* deploy to {}".format(cfg.web_url))
```

Comment:
The only thing I wonder here is if the `Notifier` class had a `__enter__()` style interface (as docker_client has) we could write:

```
with notify(destination):
  with docker_client() as cl:
    ensure_network(cl, cfg.network)
    ...
    config.save(cfg)
```

which would make this nicer to read - alternatively the inner block could be broken into a separate function but that does seem about as much work as the above suggestion.

Reply:
Would it make it nicer to read? I feel like a double nested `with` would be slightly harder to read. Second suggestion - you mean move lines 33 to 39 to own function? I think that would be clearer.

Reply:
We can also write a single with apparently (https://stackoverflow.com/a/3025119)

```
with notify(destination) as n, docker_client() as cl:
  ...
```

Reply:
but, yes otherwise just lifting 33-39 out

Reply:
Ooh, I like the multiple simultaneous contexts syntax. But I think it'd still be a bit noisy inside the `with` and the notifier isn't really a context, so I think I'll just do the latter.